File: csharp/secret-handshake/SecretHandshake.cs
using System.Collections.Generic;
using System.Linq;

public class SecretHandshake
{
    public static string[] Commands(int numberOfCommands)
    {
        var mappings = new Dictionary<int, string>
        {
            {0x1, "wink"},
            {0x2, "double blink"},
            {0x4, "close your eyes"},
            {0x8, "jump"},
        };

        var handshake = mappings.Where(x => (x.Key & numberOfCommands) != 0).Select(x => x.Value);

        return ((numberOfCommands & 10000) != 0 ? handshake.Reverse() : handshake).ToArray();
    }
}

File: swift/secret-handshake/SecretHandshake.swift
class SecretHandshake {
    let moves : Int

    var mapping : [Int:String]  = [
        0x1: "wink",
        0x2: "double blink",
        0x4: "close your eyes",
        0x8: "jump"
    ]

    init(_ moves : Int) {
        self.moves = moves
    }

    var commands : [String] {
        var output = mapping.filter({ $0.key & moves > 0 }).sorted(by: { $0 < $1 }).map({ $0.value })

        return self.moves & 0x10 > 0 ?  Array(output.reversed()) : output
    }
}

